helix-term/src/commands/lsp.rs

```diff
@@ -326,126 +331,147 @@ fn diag_picker(
+            };
+            // lsp has two ways to represent symbols (flat/nested)
+            // convert the nested variant to flat, so that we have a homogeneous list
+            let symbols = match symbols {
+                lsp::DocumentSymbolResponse::Flat(symbols) => symbols
+                    .into_iter()
+                    .map(|symbol| SymbolInformationItem {
+                        symbol,
+                        offset_encoding,
+                    })
+                    .collect(),
+                lsp::DocumentSymbolResponse::Nested(symbols) => {
+                    let mut flat_symbols = Vec::new();
+                    for symbol in symbols {
+                        nested_to_flat(&mut flat_symbols, &doc_id, symbol, offset_encoding)
+                    }
+                    flat_symbols
+                }
+            };
+            Ok(symbols)
+        })
+        .collect();
     let current_url = doc.url();
-    let offset_encoding = language_server.offset_encoding();
-
-    let future = match language_server.document_symbols(doc.identifier()) {
-        Some(future) => future,
-        None => {
-            cx.editor
-                .set_error("Language server does not support document symbols");
-            return;
-        }
-    };
 
-    cx.callback(
-        future,
-        move |editor, compositor, response: Option<lsp::DocumentSymbolResponse>| {
-            if let Some(symbols) = response {
-                // lsp has two ways to represent symbols (flat/nested)
-                // convert the nested variant to flat, so that we have a homogeneous list
-                let symbols = match symbols {
-                    lsp::DocumentSymbolResponse::Flat(symbols) => symbols,
-                    lsp::DocumentSymbolResponse::Nested(symbols) => {
-                        let doc = doc!(editor);
-                        let mut flat_symbols = Vec::new();
-                        for symbol in symbols {
-                            nested_to_flat(&mut flat_symbols, &doc.identifier(), symbol)
-                        }
-                        flat_symbols
-                    }
-                };
+    if futures.is_empty() {
+        cx.editor
+            .set_error("No Language server does support document symbols");
+        return;
+    }
 
-                let picker = sym_picker(symbols, current_url, offset_encoding);
-                compositor.push(Box::new(overlayed(picker)))
+    cx.jobs.callback(async move {
+        let mut symbols = Vec::new();
+        // TODO if one symbol request errors, all other requests are discarded (even if they're valid)
+        while let Some(mut lsp_items) = futures.try_next().await? {
+            symbols.append(&mut lsp_items);
+        }
+        let call = move |editor: &mut Editor, compositor: &mut Compositor| {
+            if symbols.is_empty() {
+                editor.set_error("No symbols available");
+                return;
             }
```

Comment:
we currently just open an empty symbol picker in this case. let's keep it this way. this is really an orthogonal concern and something that can be adressed seperatly (I think opening an empty picker is just fine actually)